draft-ietf-quic-http.md

```diff
@@ -1238,9 +1233,10 @@ MAX_PUSH_ID frame ({{frame-max-push-id}}). A client MUST treat receipt of a
 PUSH_PROMISE frame that contains a larger Push ID than the client has advertised
 as a connection error of H3_ID_ERROR.
 
-A server MUST NOT use the same Push ID in multiple PUSH_PROMISE frames. A client
-MUST treat receipt of a Push ID which has already been promised as a connection
-error of type H3_ID_ERROR.
+A server MAY use the same Push ID in multiple PUSH_PROMISE frames. If so, the
+decompressed request headers MUST be idential.  A client MUST treat receipt of a
+Push ID which has already been promised with different headers than the previous
+promise as a connection error of type H3_ID_ERROR.
```

Comment:
MUST-error requires that the client keep the request headers for the lifetime of the connection in case it ever sees that Push ID again.  Instead, I think we want to say that it MUST error if it detects a mismatch, and MUST ignore the new PUSH_PROMISE if it has not retained the previous request header set.

Reply:
> MUST-error requires that the client keep the request headers for the lifetime of the connection in case it ever sees that Push ID again.

As stated in https://github.com/quicwg/base-drafts/issues/3275#issuecomment-566379821, it is my understanding that an endpoint can determine if the PUSH_PROMISE refers to a new Push ID or a Push ID it already has received, and therefore my preference goes to stating "MUST error if there is a mismatch, or ignore the PUSH_PROMISE frame."

Reply:
No comment about the MUSTiness but H3_ID_ERROR is not appropriate for this error condition IMO. Nor is it a frame error because all the syntax was correct. The closest comparison is malformed request, which elicits a H3_GENERAL_PROTOCOL_ERROR. Alternatively connection-level H3_REQUEST_REJECTED could fit but might need some other changes in the wider document to make sure this case is accommodated for.

Reply:
I've updated the text to use the "MUST error if mismatch detected" language.